keras_cv/layers/preprocessing/jittered_resize.py

```diff
@@ -0,0 +1,238 @@
+    ```
+
+    Args:
+        target_size: A tuple repesenting the output size of images.
+        scale_factor: A tuple of two floats or a `keras_cv.FactorSampler`. For each
+            augmented image a value is sampled from the provided range.
+            This factor is used to scale the input image.
+            To replicate the results of the MaskRCNN paper pass `(0.8, 1.25)`.
+        crop_size: (Optional) the size of the image to crop from the scaled image.
+            Defaults to `target_size` when not provided.
+        bounding_box_format: The format of bounding boxes of input boxes. Refer
+            to https://github.com/keras-team/keras-cv/blob/master/keras_cv/bounding_box/converters.py
+            for more details on supported bounding box formats.
+        interpolation: String, the interpolation method. Defaults to `"bilinear"`.
+            Supports `"bilinear"`, `"nearest"`, `"bicubic"`, `"area"`, `"lanczos3"`,
+            `"lanczos5"`, `"gaussian"`, `"mitchellcubic"`.
+        seed: (Optional) integer to use as the random seed.
+    """
+
+    def __init__(
+        self,
+        target_size,
+        scale_factor,
+        crop_size=None,
+        bounding_box_format=None,
+        interpolation="bilinear",
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if not isinstance(target_size, tuple) or len(target_size) != 2:
+            raise ValueError(
+                "JitteredResize() expects `target_size` to be "
+                f"a tuple of two integers.  Received `target_size={target_size}`"
+            )
+
+        crop_size = crop_size or target_size
+        self.interpolation = keras_cv.utils.get_interpolation(interpolation)
+        self.scale_factor = keras_cv.utils.parse_factor(
+            scale_factor,
+            min_value=0.0,
+            max_value=None,
+            param_name="scale_factor",
+            seed=seed,
+        )
+        self.crop_size = crop_size
+        self.target_size = target_size
+        self.inference_resizing = layers.Resizing(
+            target_size[0],
+            target_size[1],
+            pad_to_aspect_ratio=True,
+            interpolation=interpolation,
+            bounding_box_format=bounding_box_format,
+        )
+        self.bounding_box_format = bounding_box_format
+        self.seed = seed
+        self.force_output_dense_images = True
+        self.auto_vectorize = False
+
+    def call(self, inputs, training=True):
+        if training:
+            return super().call(inputs, training)
+        else:
+            inputs = self._ensure_inputs_are_compute_dtype(inputs)
+            inputs, meta_data = self._format_inputs(inputs)
+            output = inputs
+            output["images"] = self.inference_resizing(inputs["images"])
+            return self._format_output(output, meta_data)
+
+    def get_random_transformation(self, image=None, **kwargs):
```

Comment:
As the implementation goes into public API, I feel we should acknowledge the contribution from tensorflow_models, maybe at top of the file. (Reduce IP arguments as much as possible)